app/views/external_users/claims/misc_fees/_misc_fee_fields.html.haml

```diff
@@ -10,29 +10,54 @@
       %label.form-label-bold
         = t('.fee_type')
       %a{id: "misc_fee_#{@misc_fee_count}_fee_type"}
-      = f.collection_select :fee_type_id, @claim.eligible_misc_fee_types, :id, :description, {include_blank: ''.html_safe }, {class: 'form-control js-misc-fee-type js-fee-type typeahead', style: 'width:100%'}
+      = f.collection_select :fee_type_id,
+       @claim.eligible_misc_fee_types, :id, :description,
+       {include_blank: ''.html_safe },
+       {class: 'form-control js-misc-fee-type js-fee-type js-misc-fee-calculator-fee-type typeahead', style: 'width:100%'}
 
       = validation_error_message(@error_presenter, "misc_fee_#{@misc_fee_count}_fee_type")
 
   = f.adp_text_field :quantity,
                           label: t('.quantity'),
-                          input_classes:'quantity fee-quantity form-control-1-8',
+                          hint_text: t('.quantity_hint'),
+                          input_classes:'quantity fee-quantity form-control-1-8 js-fee-quantity js-misc-fee-calculator-quantity',
                           input_type: 'number',
                           value: fee.quantity,
                           errors: @error_presenter
 
+  .js-misc-fee-calculator-effectee
+    .calculated-misc-fee
+    = f.adp_text_field :rate,
+        label: t('.rate'),
+        input_classes:'rate fee-rate js-misc-fee-calculator-rate form-input-denote__input form-control-1-4',
+        input_type: 'currency',
+        errors: @error_presenter,
+        input_readonly: f.object.is_rate_calculated?,
+        value: number_with_precision(f.object.rate, precision: 2)
+
+  .js-misc-fee-calculator-success
+    = f.hidden_field :rate_calculated, value: f.object.is_rate_calculated?
```

Comment:
As discussed, i think we can probably get rid of this method on base_fee.rb and fixed_fee.rb and rely on rails automagically generated `rate_calculated?`  - and i believe this would fix the amendable rate error on save and returning to page.